src/api/handlers.go

```diff
@@ -404,22 +403,11 @@ func KillServer(w http.ResponseWriter, r *http.Request) {
 }
 
 func CheckServer(w http.ResponseWriter, r *http.Request) {
-	resp := map[string]string{}
-
 	defer func() {
-		WriteResponse(w, resp)
+		WriteResponse(w, factorio.GetFactorioServer())
 	}()
 
 	w.Header().Set("Content-Type", "application/json;charset=UTF-8")
-	var server = factorio.GetFactorioServer()
-	if server.GetRunning() {
-		resp["status"] = "running"
-		resp["port"] = strconv.Itoa(server.Port)
-		resp["savefile"] = server.Savefile
-		resp["address"] = server.BindIP
-	} else {
-		resp["status"] = "stopped"
-	}
```

Comment:
You have removed resp["status"], but the frontend is still using it. This currently results in most of the frontend is thinking that the server is currently in the wrong state (no state). Examples: "Mods" thinking the server is always running and "Console" is always thinking the server is offline. You should also use `server.running` and remove all references to `status` in the frontend.